lib/launchers/ocm.rb

```diff
@@ -0,0 +1,238 @@
+    def default_opts(service_name)
+      return  conf[:services, service_name.to_sym]
+    end
+
+    def to_seconds(string)
+      regex_m = /^(\d+)\s*(m|min|minutes|mins)+$/
+      regex_h = /^(\d+)\s*(h|hour|hours|hrs)+$/
+      regex_d = /^(\d+)\s*(d|day|days)+$/
+      regex_w = /^(\d+)\s*(w|week|weeks|wks)+$/
+      case string
+      when regex_m
+        return string.match(regex_m)[1].to_i * 60
+      when regex_h
+        return string.match(regex_h)[1].to_i * 60 * 60
+      when regex_d
+        return string.match(regex_d)[1].to_i * 24 * 60 * 60
+      when regex_w
+        return string.match(regex_w)[1].to_i * 7 * 24 * 60 * 60
+      else
+        raise "Cannot convert '#{string}' to seconds!"
+      end
+    end
+
+    # create a json which specifies OSD cluster
+    # in the future we plan to move the logic into the script 'osd-provision.sh'
+    def generate_json(name)
+      json_data = {
+        "name" => name,
+        "managed" => true,
+        "multi_az" => false,
+        "byoc" => false
+      }
+
+      if @multi_az
+        json_data.merge!({"multi_az" => @multi_az})
+      end
+
+      if @region
+        json_data.merge!({"region" => {"id" => @region}})
+      end
+
+      if @version
+        json_data.merge!({"version" => {"id" => "openshift-v#{@version}"}})
+      end
+
+      if @nodes
+        json_data.merge!({"nodes" => {"compute" => @nodes.to_i}})
+      end
+
+      if @lifespan
+        expiration = Time.now + to_seconds(@lifespan)
+        json_data.merge!({"expiration_timestamp" => expiration.strftime("%Y-%m-%dT%H:%M:%SZ")})
+      end
+
+      if @aws_account_id && @aws_access_key && @aws_secret_key
+        json_data.merge!({"aws" => {"account_id":@aws_account_id, "access_key_id":@aws_access_key, "secret_access_key":@aws_secret_key}})
+        json_data.merge!({"byoc" => true})
+      end
+
+      return json_data.to_json
+    end
+
+    # download the script 'osd-provision.sh' which takes care of the OSD installation/uninstallation
+    def download_osd_script
+      osd_repo_uri = ENV['GIT_OSD_URI'] || 'https://gitlab.cee.redhat.com/mk-bin-packing/mk-performance-tests.git'
+      osd_repo_dir = File.join(Dir.tmpdir, 'osd_repo')
+      FileUtils.rm_rf(osd_repo_dir)
+      git = BushSlicer::Git.new(uri: osd_repo_uri, dir: osd_repo_dir)
+      git.clone
+      osd_script = File.join(osd_repo_dir, 'scripts', 'osd-provision.sh')
+      if !File.exists?(osd_script)
+        raise "Cannot find #{osd_script}"
+      end
+      return osd_script
+    end
+
+    def shell(cmd, output = nil)
+      if output
+        res = Host.localhost.exec(cmd, single: true, stderr: :stdout, stdout: output, timeout: 3600)
+      else
+        res = Host.localhost.exec(cmd, single: true, timeout: 3600)
+      end
+      if res[:success]
+        return res[:response]
+      else
+        raise "Error when executing '#{cmd}'. Response: #{res[:response]}"
+      end
+    end
+
+    def get_last_line(text)
```

Comment:
for small texts you can just do `str.lines.last` and not need a special method

Reply:
fixed